src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildStatusMacro.java

```diff
@@ -43,7 +40,8 @@ public String evaluate(Run<?,?> run, FilePath workspace, TaskListener listener,
             throws MacroEvaluationException, IOException, InterruptedException {
 
         // In the case of pipeline jobs, if the status hasn't been set to a non-null value, then it is considered "success"
-        if(run instanceof WorkflowRun && null == run.getResult()) {
+        if (!(run instanceof AbstractBuild) && null == run.getResult()) {
+            // TODO this makes little sense. Why not just use Building as below?

```

Comment:
TBD, but anyway suffices to check for non-`AbstractBuild`s.

Reply:
This was a specific request so that email-ext stuff would work during a build.

Reply:
Right, but would that not be covered by the check for a null result ~ `Building` below this already?

Reply:
They didn't want 'Building' they wanted a Success, but the job hasn't completed yet. So, it was an intermediate 'Success' they were looking for. See https://issues.jenkins-ci.org/browse/JENKINS-44322